src/main/java/frc/robot/states/AmpState.java

```diff
@@ -0,0 +1,56 @@
+package frc.robot.states;
+
+import com.pathplanner.lib.auto.AutoBuilder;
+import com.pathplanner.lib.util.GeometryUtil;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Transform2d;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import frc.robot.Constants;
+import frc.robot.swerve.SwerveDrive;
+import lib.Utils;
+
+public class AmpState implements ScoreState {
+
+    @Override
+    public Command driveToClosestOptimalPoint() {
+        return Commands.defer(
+                () -> {
+                    Pose2d ampPose = Constants.AMP_POSE;
```

Comment:
I think it'll be clearer to save `AMP_POSE` as a `Translation2d` and then add a rotation component here, based on the desired direction. Currently, it's not obvious whether `AMP_POSE` is initialized with the shooter or intake side towards the Amp.

Reply:
ad3ab56

Reply:
I meant something else, the rotation isn't determined until further in the code, so creating a rotation of 90° here is not as clear as creating it in the constants file.

Reply:
I meant something else, the rotation isn't determined until further in the code, so creating a rotation of 90° here is not as clear as creating it in the constants file.

Reply:
9360666